src/sensor_drivers/bmi08_defs.h

```diff
@@ -819,6 +819,10 @@ struct bmi08_sensor_data
 
     /*! Z-axis sensor data */
     int16_t z;
+
+    uint32_t norm_squared (){
+        return x*x + y*y + z*z;
+    }
 };
```

Comment:
Where are those helper functions used ? If we only need the norm in the FSM or high level modules, then it might be better to implement this function in the `Vector3` struct in `data.h/cpp`. Also as it is a common function to any 3D data, implementing it only once would be better. Also why the need of an `uint32_t` return type ?